Reading: Update MegaReadingImporter to 8/20/19 template format

kevinrobinson opened the pull request:
Part of https://github.com/studentinsights/studentinsights/pull/2523.

# Who is this PR for?
Elementary teachers, reading specialists and district staff in Somerville

# What problem does this PR fix?
Updates `MegaReadingImporter` to work with the 8/20/19 template format, with tests.  This will enable using this manually for some historical spring data that was entered in a slightly different format, and merges this in for use as part of automated imports in September more generally.

# What does this PR do?
Builds on @edavidsonsawyer's awesome work in https://github.com/studentinsights/studentinsights/pull/2523, taking `MegaReadingProcessor` and the related fixture and spec.  It adds a few small changes to tracking and how headers are handled, and adds some comments as well.

@edavidsonsawyer This addresses some of the comments in https://github.com/studentinsights/studentinsights/pull/2523, so I think a step forward there is to merge this into that branch, and then work forward on the importer that makes use of this on a folder.  I'm happy to chat more over on that issue or over email.

Peeling this off and merging it in so we can get some historical data imported now, and then we can finish off the importer component separately, since that won't be needed for another few weeks.

# Checklists
*Does this PR use tests to help verify we can deploy these changes quickly and confidently?*
+ [x] Included specs for changes
+ [x] Manual testing made more sense here

edavidsonsawyer commented:
@kevinrobinson this is great! I'll merge this into the other branch and we can focus on the importer there.

kevinrobinson commented:
Added some commits updating the UI debug tool (use `gradeThen` when determining benchmark thresholds, including coverage of active students, related specs).

Also fixed some bugs in the processor code (some bugs in the keys matching the import templates, periods not matching the import column keys).

kevinrobinson commented:
selfie cc @edavidsonsawyer
